integrationTests/chainSimulator/bridge/esdtSafe_test.go

```diff
@@ -0,0 +1,518 @@
+
+		// create random receiver addresses in a shard
+		receiver, _ := cs.GenerateAndMintWalletAddress(receiverShardId, chainSim.InitialAmount)
+		receiverNonce := uint64(0)
+
+		// execute operations received from sovereign chain
+		// expecting the token to be minted in esdt-safe contract with the same properties and transferred to receiver address
+		// -------------
+		// for (dynamic) SFT/MetaESDT the contract will create one more token and keep it forever
+		// because if the same token is received 2nd time, the contract will just add quantity, not create different token
+		txResult := executeOperation(t, cs, bridgeData.OwnerAccount.Wallet, receiver.Bytes, &bridgeData.OwnerAccount.Nonce, bridgeData.ESDTSafeAddress, []chainSim.ArgsDepositToken{token}, wallet.Bytes, nil)
+		chainSim.RequireSuccessfulTransaction(t, txResult)
+		receivedToken := chainSim.ArgsDepositToken{
+			Identifier: tokensMapper[token.Identifier],
+			Nonce:      token.Nonce,
+			Amount:     token.Amount,
+			Type:       token.Type,
+		}
+		if isSftOrMeta(receivedToken.Type) {
+			chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), esdtSafeAddr, big.NewInt(1))
+		} else {
+			chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), esdtSafeAddr, big.NewInt(0))
+		}
+
+		waitIfCrossShardProcessing(cs, esdtSafeAddrShard, receiverShardId)
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), receiver.Bech32, receivedToken.Amount)
+
+		// --------------------------------------------
+		// deposit back 1 token, expect token to be burned and supply burned is 1
+		depositAmount := big.NewInt(1)
+		depositToken := chainSim.ArgsDepositToken{
+			Identifier: receivedToken.Identifier,
+			Nonce:      receivedToken.Nonce,
+			Amount:     depositAmount,
+			Type:       receivedToken.Type,
+		}
+		txResult = deposit(t, cs, receiver.Bytes, &receiverNonce, bridgeData.ESDTSafeAddress, []chainSim.ArgsDepositToken{depositToken}, wallet.Bytes)
+		chainSim.RequireSuccessfulTransaction(t, txResult)
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(receivedToken), receiver.Bech32, big.NewInt(0).Sub(receivedToken.Amount, big.NewInt(1)))
+
+		waitIfCrossShardProcessing(cs, esdtSafeAddrShard, receiverShardId)
+
+		expectedAmount := big.NewInt(0)
+		if isSftOrMeta(receivedToken.Type) {
+			// for (dynamic) SFT/MetaESDT, the contract should always have 1 esdt
+			expectedAmount = big.NewInt(1)
+		}
+		chainSim.RequireAccountHasToken(t, cs, getTokenIdentifier(depositToken), esdtSafeAddr, expectedAmount)
+
+		_ = cs.GenerateBlocks(1)
+		tokenSupply, err := cs.GetNodeHandler(esdtSafeAddrShard).GetFacadeHandler().GetTokenSupply(getTokenIdentifier(depositToken))
+		require.Nil(t, err)
+		require.NotNil(t, tokenSupply)
+		require.Equal(t, depositAmount.String(), tokenSupply.Burned)
```

Comment:
[question]
How come this token's supply is always burned, yet the contract keeps one extra for `for (dynamic) SFT/MetaESDT`?
Is that kept one a different one that the contract mints?

Reply:
This burn is just related to deposit amounts, what is deposited will be burned.
But correct, the contract will mint one more `(dynamic) SFT/MetaESDT` for a specific ID-nonce and will keep it forever so he will be able to add quantity later